### PR TITLE
Strict check for URL params

### PR DESCRIPTION
**Pull request by sarveshbathija**
By submitting a PR to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/auth0/.github/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

Current code checks if string code=, state=, error=.  This causes issues if there is a substring that contains these strings.  scode=, estate=, abceferror=.  These all pass erroneously thinking it is an auto0 application.  My app was failing because state was a substring of one of my url params.

This fix is to strictly check if there is a url param that exists and also is populated.

### Testing

To test, currently functionality should still work when there is a code=, state= in the URL.  However it should NOT continue now if there is a url with abccode=, defstate= or ghijerror= in the URL.  All current tests pass with this change.

I also added a test to check that it does not call the redirect call back if the url params are `abccode=` or `xyzstate=`
- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x ] All active GitHub checks for tests, formatting, and security are passing
- [x ] The correct base branch is being used, if not the default branch

### CODE REVIEW

**Review by frederikprijck — COMMENTED** [review]
Thanks for the change. Even tho I think this is a great improvement, I do not think we can merge this as IE11 has no support for URLSearchParams.

I would be happy to merge this once we cut v2, but for v1 I believe you will need to use `skipRedirectCallback`:

```ts
AuthModule.forRoot({
  skipRedirectCallback: location.search.includes('abccode=') || location.search.includes('xyzstate=')
})
```
Definetly not ideal, and I agree this PR would be the better change. But we can't drop support for IE11, and I believe we shouldn't introduce any polyfills if we can work around it.

We are working on a major improvement of Auth0-SPA-JS, where support for IE11 is dropped. This will be followed by a major of our Angular SDK, where we can safely use URLSearchParams.

**Comment by frederikprijck** [comment]
Just a quick update ... Our SPA-JS SDK v2 has gone in beta, dropping support for IE11 and the like.
We expect our Angular SDK to follow with a new beta as well, but not sooner then once SPA-JS has hit GA.

Will merge this PR in a beta branch once we are at that point. Thanks for this change 👍

**Comment by sarveshbathija** [comment]
@frederikprijck sounds great.  Thx!

File: projects/auth0-angular/src/lib/auth.service.ts [hunk 1]
```diff
@@ -404,9 +404,10 @@ export class AuthService<TAppState extends AppState = AppState>
   private shouldHandleCallback(): Observable<boolean> {
     return of(location.search).pipe(
       map((search) => {
+        const searchParams = new URLSearchParams(search);
         return (
-          (search.includes('code=') || search.includes('error=')) &&
-          search.includes('state=') &&
+          (!!searchParams.get('code') || !!searchParams.get('error')) &&
+          !!searchParams.get('state') &&
```
Comment (frederikprijck) [inline projects/auth0-angular/src/lib/auth.service.ts]:
```suggestion
          (searchParams.has('code') || searchParams.has('error')) &&
          searchParams.has('state') &&
```

**Review by frederikprijck — CHANGES_REQUESTED** [review]

**Review by frederikprijck — DISMISSED** [review]
Thanks 🙏

**Pull request merged**